pkgs/misc/cups/drivers/kyodialog/default.nix

```diff
@@ -0,0 +1,109 @@
+{ lib
+, stdenv
+, fetchzip
+, cups
+, autoPatchelfHook
+, python3Packages
+
+# Sets the default paper format: use "EU" for A4, or "Global" for Letter
+, region ? "Global"
+# optional GUI, quite redundant to CUPS admin web GUI
+, withQtGui ? false
+, qt5
+}:
+
+# Open issues:
+#
+# Printing in grayscale mode (e.g. the test page) generates a warning
+# > [Job 59] Grayscale/monochrome printing requested for this job but Poppler is not able to convert to grayscale/monochrome PostScript.
+# > [Job 59] Use \"pdftops-renderer\" option (see cups-filters README file) to use Ghostscript or MuPDF for the PDF -> PostScript conversion.
+#
+# This is related to https://github.com/OpenPrinting/cups-filters/issues/169.
+
+let
+  platform =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "amd64"
+    else if stdenv.hostPlatform.system == "i686-linux" then "i386"
+         else throw "Unsupported system: ${stdenv.hostPlatform.system}";
```

Comment:
Please use an attr instead.

Reply:
> Can we just remove those very outdated and insecure libraries [openssl, curl]? Also with current laws such things must be opt-in rather than opt-out.

The problem is that the libs are statically linked into the binary. I only noticed it because the binary is surprisingly large (2.1 MB). I don't know of a way to remove statically linked libraries from an ELF file.
* As we know the exact *.a file from openssl Kyocera used, we could `nop` out everything. curl seems manually compiled ("-DEV" version string), so we don't know the linked *.a file.
* We could binary-patch the `https://www.google-analytics.com/batch` URL to e.g. `http://localhost.invalid/` (and fill in the remainder with `\0`), so that it cannot upload the usage statistics, even if it was enabled.
* This does not reduce the attack surface due to a lot of dead code (there are probably many nice gadgets in curl and openssl).

Re. opt-in/out: it looks to me that Kyocera implemented it correctly, i.e. the upload must be enabled in the config file and this parameter is also asked for every time the GUI is launched. Now, if a user doesn't use the GUI, s/he never gets asked to opt-in to Google Aanalytics and the config is never enabled.



> > * What should be the default paper size?
> >   `region = "Global"` (a misnomer) is Letter, while `region = "EU"` (also a misnomer) is A4. (This can be seen when diffing the 2 .deb files, which are exactly the same but the default page size in the PPD files)
> 
> I am biased and would choose A4 which also sometimes happens when your display language is German but often the default is letter.

I am biased too, but I think A4 is more common. I changed the default to A4 ("EU").